/lidar_mirror_fov_reshaper_calibration: # for any further parameter description, please checkout/build the documentation
  ros__parameters:
    laser_scanner:
      topic: "cloud"
      frame_id: "cloud"
      angle_min: -135 # deg, int
      angle_max: 135 # deg, int
      use_pointcloud_input: false # if false, laserscan input is being used

    visualization:
      transformed_cloud_all: true
      transformed_cloud_rm: true
      transformed_cloud_lm: true
      transformed_cloud_front: true
      normal_vectors: true
      optimized_planes_all: true
      optimized_plane_rm: true
      optimized_plane_lm: true
      optimized_plane_opt_plane: true
      optimization_plane_box: true 

    optimization:
      filter_method: 0
      filter_dist_threshold: 0.1
      interpolation_window: 0 
      averaging_n_clouds: -1 
      intensity_threshold_percentage: 0.1
      buffer_size: 3500
      epsabs: 1e-4
      stepsize: 1e-3
      iter_max: 1e3
      adaptive_stepsize: false
      opt_mirror_orientation: true
      opt_mirror_support_vec: false
      write_optimized_params: false
      write_optimization_history: false
      optimization_history_file_: "lmfrc_history.csv"
      optimized_params_meta_file: "lmfrc_metadata.csv"
      optimized_params_file: "lmfrc_results.csv"
      optimized_params_out_dir: "/my_out_dir/"
      verbose: 1 
      evaluation_no_batches: 1
      optimization_mirror_orientation:
        opt_all: false 
        opt_mirror_svs: false 
        opt_mirror_nvs: false 
        opt_plane_sv: false
        opt_osg_settings: false
        right_mirror:
          support_vec:
            x: false
            y: false
            z: false
          normal_vec:
            x: true
            y: true
            z: true
        left_mirror:
          support_vec:
            x: false
            y: false
            z: false
          normal_vec:
            x: true
            y: true
            z: true
        calibration_plane:
          support_vec:
            x: true
            y: true
            z: true
          normal_vec:
            x: true
            y: true
            z: true

    calibration_plane:
      support_vec: [0.15, 0.0, -0.35]
      helper_p1: [0.15, 0.35, 0.0]
      helper_p2: [0.15, -0.2, 0.0]
      helper_p3: [0.15, 0.0, 0.1]

    mirror_front:
      start_angle: -45
      end_angle: 45

    mirror_left:
      mirror_safety_bufferzone_size: 5
      auto_define_start_angle: false
      auto_define_end_angle: false
      auto_define_angle_mode: 0
      start_angle: 97
      end_angle: 128
      helper_p1: [-0.018, 0.057, -0.01846]
      helper_p2: [-0.042, 0.045, 0.01846]
      helper_p3: [-0.015, 0.057, 0.01475]
      support_vec: [0.0, 0.094, 0.0]
      normal_vec: [0.45, -0.33, -0.403583]

    mirror_right:
      mirror_safety_bufferzone_size: 4
      auto_define_start_angle: false
      auto_define_end_angle: false
      auto_define_angle_mode: 0 
      start_angle: -131
      end_angle: -97
      helper_p1: [-0.018, -0.057, -0.01846]
      helper_p2: [-0.042, -0.045, 0.01846]
      helper_p3: [-0.015, -0.057, 0.01475]
      support_vec: [0.0, -0.095, 0.0]
      normal_vec: [0.4, 0.29, -0.33]
